cpp/src/Ice/Instance.cpp

```diff
@@ -1227,6 +1228,8 @@ IceInternal::Instance::Instance(const CommunicatorPtr& communicator, const Initi
             throw InitializationException(__FILE__, __LINE__, "The value for Ice.ToStringMode must be Unicode, ASCII or Compat");
         }
 
+        const_cast<bool&>(_acceptClassCycles) = _initData.properties->getPropertyAsInt("Ice.AcceptClassCycles") > 0;
```

Comment:
While not strictly backwards compatible, I think in this case it should be disabled by default since your and send unsolicited cycles to a server and cause silent leaks.